### PR TITLE
[TPU] added numbers for TPU FLOPS

### PR DESCRIPTION
**Pull request by gkroiz**
PR to add FLOPS numbers for benchmarking when using TPUs.

Follow up to https://github.com/Lightning-AI/lit-gpt/pull/147#discussion_r1230009025

### CODE REVIEW

**Comment by gkroiz** [comment]
cc @jekbradbury

File: lit_parrot/speed_monitor.py [hunk 1]
```diff
@@ -57,6 +57,19 @@
     "quadro rtx 5000": {"32-true": 11.2e12, "16-true": 89.2e12, "16-mixed": 89.2e12},
 }
 
+TPU_AVAILABLE_FLOPS={
+    # flop count for each TPU generation is the same for all precisions
+    # since bfloat16 precision is always used for performing matrix operations
+    # for more info: https://cloud.google.com/tpu/docs/bfloat16#choosing_bfloat16
+
+    # source: https://arxiv.org/pdf/1907.10701.pdf
+    "V2": 45e12,
```
Comment (carmocca) [inline lit_parrot/speed_monitor.py:66]: keys need to be lowercase

File: lit_parrot/speed_monitor.py [hunk 2]
```diff
@@ -57,6 +57,19 @@
...
+    # source: https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#tpu_v3
+    "V3": 123e12,
```
Comment (carmocca) [inline lit_parrot/speed_monitor.py:68]: Shouldn't split the specific configs (v4-8, v4-128, ...)? As I had in https://github.com/Lightning-AI/lit-gpt/pull/172/files#diff-d83a788c6cc66e57022968bc902a9c15ff5e29f19e55f0f0aaadaae37f132cb8R59-R61

File: lit_parrot/speed_monitor.py [hunk 3]
```diff
@@ -89,8 +102,15 @@ def get_flops_available(device: torch.device, precision: str) -> Optional[float]
                     "MFU cannot be calculated and reported."
                 )
     elif device.type == "xla":
-        # TODO
-        ...
+        from torch_xla.experimental import tpu
+        try:
+            tpu_generation = tpu.get_tpu_env()['TYPE']
+            return TPU_AVAILABLE_FLOPS[tpu_generation]
```
Comment (carmocca) [inline lit_parrot/speed_monitor.py:108]: Use xenv as in https://github.com/Lightning-AI/lit-gpt/pull/172/files#diff-d83a788c6cc66e57022968bc902a9c15ff5e29f19e55f0f0aaadaae37f132cb8R98-R100?

**Comment by gkroiz** [comment]
@carmocca please take a look at the code snippet below:
```
>>> from torch_xla.experimental import tpu
>>> tpu.get_tpu_env()
{'ACCELERATOR_TYPE': 'v4-8', 'AGENT_BOOTSTRAP_IMAGE': 'REDACTED', 'ALT': 'false', 'CHIPS_PER_HOST_BOUNDS': '2,2,1', 'COLLECTD_DOCKER_URL': 'REDACTED', 'CONSUMER_PROJECT_ID': 'mlperf-high-priority-project', 'CONSUMER_PROJECT_NUMBER': '903354779218', 'CONTROL_MESSAGE_SOURCE': 'pubsub', 'ENABLE_ICI_RESILIENCY': 'false', 'ENABLE_IMPROVED_REROUTE_ALLREDUCE_STRATEGY': 'false', 'ENABLE_MEMCACHED': 'false', 'FLUENTD_DOCKER_URL': 'REDACTED', 'HEALTH_AGENT_DOCKER_URL': 'REDACTED', 'HOST_BOUNDS': '1,1,1', 'INFERENCE_MODE': 'false', 'INJECT_SLICE_BUILDER_FAULT': '', 'INTERNAL': 'true', 'MAINTENANCE_ACTION_FLAG': 'unhealthy-maintenance', 'MEMCACHED_DOCKER_URL': '', 'MONITORING_AGENT_DOCKER_URL': 'REDACTED', 'NODE_ID': 'REDACTED', 'PREEMPTIBLE': 'false', 'REPORTING_MODE': 'pubsub-and-metadata', 'RUNTIME_MONITOR_DOCKER_URL': 'REDACTED', 'RUNTIME_VERSION': 'REDACTED', 'RUNTIME_VERSION_CHANGER_DOCKER_URL': 'REDACTED', 'SERVICE_NAME': 'tpu.googleapis.com', 'SOURCE': '', 'TOPOLOGY': '2x2x1', 'TPU_CHIPS_PER_PROCESS_BOUNDS': '2,2,1', 'TPU_PROCESS_BOUNDS': '1,1,1', 'TPU_TOPOLOGY_ALT': 'false', 'TPU_TOPOLOGY_WRAP': 'false,false,false', 'TYPE': 'V4', 'UID': 'REDACTED', 'USE_DIRECT_PATH': 'false', 'WORKER_ID': '0', 'WRAP': 'false,false,false', 'ZONE': 'us-central2-b'}
```

If the key is `ACCELERATOR_TYPE`, then the value is `v4-8`. If the key is `TYPE`, the value is `V4`. Do we also want the number of cores (i.e, the num 8 in v4-8)?

**Comment by carmocca** [comment]
Oh yes, you're right. We multiply this number by the world size, so we don't want the number of cores: https://github.com/Lightning-AI/lit-gpt/blob/main/lit_parrot/speed_monitor.py#L223

**Review by carmocca — APPROVED** [review]

File: lit_parrot/speed_monitor.py [hunk 4]
```diff
@@ -57,17 +57,16 @@
...
 
-TPU_AVAILABLE_FLOPS={
+TPU_AVAILABLE_FLOPS = {
     # flop count for each TPU generation is the same for all precisions
     # since bfloat16 precision is always used for performing matrix operations
     # for more info: https://cloud.google.com/tpu/docs/bfloat16#choosing_bfloat16
-
     # source: https://arxiv.org/pdf/1907.10701.pdf
-    "V2": 45e12,
+    "v2": 45e12,
```
Comment (gkroiz) [inline lit_parrot/speed_monitor.py:65]: from the output in https://github.com/Lightning-AI/lit-gpt/pull/186#issuecomment-1602963959, shouldn't the `v` be `V`?

File: lit_parrot/speed_monitor.py [hunk 5]
```diff
@@ -57,17 +57,16 @@
...
     # source: https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#tpu_v3
-    "V3": 123e12,
+    "v3": 123e12,
```
Comment (gkroiz) [inline lit_parrot/speed_monitor.py:67]: same comment as above

File: lit_parrot/speed_monitor.py [hunk 6]
```diff
@@ -57,17 +57,16 @@
...
     # source: https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#tpu_v4
-    "V4": 275e12,
+    "v4": 275e12,
```
Comment (gkroiz) [inline lit_parrot/speed_monitor.py:69]: same comment as above

File: lit_parrot/speed_monitor.py (hunk 4 above)
Comment (carmocca) [reply lit_parrot/speed_monitor.py:65]: I added .lower() to be case invariant

Comment (gkroiz) [reply lit_parrot/speed_monitor.py:65]: Ah thanks for pointing this out, I didn't notice

**Pull request merged**